src/Bicep.Core/Emit/TemplateEmitter.cs

```diff
@@ -70,8 +82,8 @@ private EmitResult EmitOrFail(Action write)
             {
                 return new EmitResult(EmitStatus.Failed, diagnostics);
             }
-
-            write();
+            
+            write(this.CalculateTemplateHash(assemblyVersion));
```

Comment:
There's quite a lot going on in this line - I think worth breaking it out into two to make it easier to read:
```csharp
var templateHash = this.CalculateTemplateHash(assemblyVersion);

write(templateHash);
```